app/view_models/auction_list_item.rb

```diff
@@ -0,0 +1,193 @@
+class AuctionListItem
+  attr_reader :auction, :current_user
+
+  def initialize(auction:, current_user:)
+    @auction = auction
+    @current_user = current_user
+  end
+
+  def title
+    auction.title
+  end
+
+  def id
+    auction.id
+  end
+
+  def html_summary
+    return '' if auction.summary.blank?
+    markdown.render(auction.summary).html_safe
+  end
+
+  def label
+    status_presenter.label
+  end
+
+  def label_class
+    status_presenter.label_class
+  end
+
+  def capitalized_formatted_type
+    if auction.type == "single_bid"
+      'single-bid'
+    else
+      'multi-bid'
+    end.capitalize
+  end
+
+  def eligibility_label
+    if for_small_business?
+      'Small-business only'
+    else
+      'SAM.gov only'
+    end
+  end
+
+  def available?
+    auction_status.available?
+  end
+
+  def current_user_can_bid?
+    if current_user.is_a?(Guest)
+      true
+    elsif !current_user.sam_accepted?
+      false
+    elsif for_small_business? && current_user.small_business?
+      true
+    elsif for_small_business? && !current_user.small_business?
+      false
+    else # not for small business
+      true
+    end
+  end
+
+  def bid_button_partial
+    if current_user_can_bid?
+      'auctions/bid_button'
+    else
+      'components/null'
```

Comment:
Is this a convention to render an empty partial? It seems a little odd to me. Adding the conditional logic `current_user_can_bid?` in the template would take up more space and lead to some redundancy, but seems worth it for easier readability.